### ISSUE TITLE
HAMA forcing TVDB Metadata, even with [anidb-123] in match or folder name

### ISSUE THREAD

**Issue by sheik124** [ZeroQI/Hama.bundle#319]
![image](https://user-images.githubusercontent.com/2312934/59121276-a49c1b00-8925-11e9-9cc8-5b76ed4e0a37.png)

Here's one example, Outlaw Star. Demon of the Water Planet was only Ep 16 in the US (and on TVDB). It's "S1" on anidb, and my files are named according to the anidb listings.

![image](https://user-images.githubusercontent.com/2312934/59121341-d57c5000-8925-11e9-908b-6ef7f32dada9.png)

The strange thing is that for a VERY BRIEF moment, S00E01 actually showed up in Plex as "Demon of the Water Planet" - it is not in the file name either, so I know it wasn't that. The correct Episode title for S00E02 in this screenshot was added manually (ie. edited).

I can provide logs or other info but I'm not sure where to start! Would also probably look through them before uploading. I mentioned earlier that certain series just stopped pulling new metadata entirely.

![image](https://user-images.githubusercontent.com/2312934/59121483-2ee47f00-8926-11e9-85c7-5142bed65d7c.png)

eg. JJBA: Golden Wind

I looked through the troubleshooting instructions, and the first suggestion is actually deleting and re-creating the library. I can't do that, I have a few years of custom posters/thumbnails/backgrounds that I don't want to lose. Very few of them are in my media folders as poster.jpg etc, most of them were "Paste URL" and I'd lose them.

**Comment by sheik124** [comment]
![image](https://user-images.githubusercontent.com/2312934/59121751-001ad880-8927-11e9-9a33-3620bd8adfdb.png)
Still shows as the Sort Title, it's pulling the anidb info for sure. I'm not sure what it's doing with it or why it's stuck in [tvdb-] mode.

**Comment by sheik124** [comment]
What are the relevant log files here?

I've added a few new folders with and without [anidb-...] that just aren't showing up at all, although I think that means ASS has stopped working.

Is there a way to force anidb metadata as the top priority? I have it first in almost every option

![image](https://user-images.githubusercontent.com/2312934/59164248-862a5100-8ad8-11e9-9f56-1f1d0a735d36.png)

**Comment by sheik124** [comment]
Filelist.log for a series not showing up at all in my library.

```
=============================================================================================================================================================
Library: 'no valid X-Plex-Token.id', root: 'E:\Media\Anime', path: 'Spice and Wolf [anidb-5406]', files: '24', dirs: '0'
Plex scan start: 2019-06-09 17:09:25,329000
plexignore_files: '[]', plexignore_dirs: '[]'
=============================================================================================================================================================
# File: 'Spice and Wolf [anidb-5406]\57690.jpg' not in 'VIDEO_EXTS'
# File: 'Spice and Wolf [anidb-5406]\106986.jpg' not in 'VIDEO_EXTS'
# File: 'Spice and Wolf [anidb-5406]\233308.jpg' not in 'VIDEO_EXTS'
# File: 'Spice and Wolf [anidb-5406]\1347144_Latvian_ShowMasterKeyArt_e173306c-a478-e711-8175-020165574d09.jpg' not in 'VIDEO_EXTS'
[file] Spice and Wolf [anidb-5406]\[Beatrice-Raws] Spice and Wolf NCED.mkv
[file] Spice and Wolf [anidb-5406]\[Beatrice-Raws] Spice and Wolf NCOP.mkv
# File: 'Spice and Wolf [anidb-5406]\cov-5151-spice-and-wolf-volume-1-japanese.jpg' not in 'VIDEO_EXTS'
# File: 'Spice and Wolf [anidb-5406]\cov-5151-spice-and-wolf-volume-1-japanese.xcf' not in 'VIDEO_EXTS'
# File: 'Spice and Wolf [anidb-5406]\desktop.ini' not in 'VIDEO_EXTS'
# File: 'Spice and Wolf [anidb-5406]\poster.jpg' not in 'VIDEO_EXTS'
[file] Spice and Wolf [anidb-5406]\Spice and Wolf - 01 - Wolf and Best Clothes [MTBB-CTR-Baal-Cman].mkv
[file] Spice and Wolf [anidb-5406]\Spice and Wolf - 02 - Wolf and Distant Past [MTBB-CTR-Baal-Cman].mkv
[file] Spice and Wolf [anidb-5406]\Spice and Wolf - 03 - Wolf and Business Talent [MTBB-CTR-Baal-Cman].mkv
[file] Spice and Wolf [anidb-5406]\Spice and Wolf - 04 - Wolf and Her Helpless Partner [MTBB-CTR-Baal-Cman].mkv
[file] Spice and Wolf [anidb-5406]\Spice and Wolf - 05 - Wolf and Lovers' Quarrel [MTBB-CTR-Baal-Cman].mkv
[file] Spice and Wolf [anidb-5406]\Spice and Wolf - 06 - Wolf and Silent Farewell [MTBB-CTR-Baal-Cman].mkv
[file] Spice and Wolf [anidb-5406]\Spice and Wolf - 07 - Wolf and Virtuous Scales [MTBB-CTR-Baal-Cman].mkv
[file] Spice and Wolf [anidb-5406]\Spice and Wolf - 08 - Wolf and the Shepherd's Lamb [MTBB-CTR-Baal-Cman].mkv
[file] Spice and Wolf [anidb-5406]\Spice and Wolf - 09 - Wolf and the Swirling Plot [MTBB-CTR-Baal-Cman].mkv
[file] Spice and Wolf [anidb-5406]\Spice and Wolf - 10 - Wolf and the Biggest Secret Scheme [MTBB-CTR-Baal-Cman].mkv
[file] Spice and Wolf [anidb-5406]\Spice and Wolf - 11 - Wolf and a Group of Youngsters [MTBB-CTR-Baal-Cman].mkv
[file] Spice and Wolf [anidb-5406]\Spice and Wolf - 12 - Wolf and a New Beginning [MTBB-CTR-Baal-Cman].mkv
[file] Spice and Wolf [anidb-5406]\Spice and Wolf - S1 - Wolf and a Tail of Happiness [MTBB-CTR-Baal-Cman].mkv
# File: 'Spice and Wolf [anidb-5406]\spice_and_wolf_season_1_v2_1_by_noavalons_dcwoluc.ico' not in 'VIDEO_EXTS'
=============================================================================================================================================================
Plex scan end: 2019-06-09 17:09:25,335000

```

Scanner.log

```
Library: 'no valid X-Plex-Token.id', root: 'E:\Media\Anime', path: 'Spice and Wolf [anidb-5406]', files: '15', dirs: '0'
Plex scan start: 2019-06-09 17:09:25,337000
=============================================================================================================================================================
Forced ID (series folder) - source: "anidb", id: "5406"
-------------------------------------------------------------------------------------------------------------------------------------------------------------
misc_count: {'and': 28, 'Lamb': 1, 'Group': 1, 'Silent': 1, 'Her': 1, 'Plot': 1, 'NCOP': 1, 'S1': 1, 'Youngsters': 1, 'Secret': 1, 'Biggest': 1, 'New': 1, 'Scheme': 1, '02': 1, '03': 1, '01': 1, '06': 1, '07': 1, '04': 1, '05': 1, '08': 1, '09': 1, "Shepherd's": 1, 'Farewell': 1, 'NCED': 1, 'Clothes': 1, 'Beginning': 1, 'Spice': 15, 'a': 3, 'Talent': 1, 'Virtuous': 1, 'Swirling': 1, 'Wolf': 28, 'the': 3, 'Quarrel': 1, '11': 1, '10': 1, '12': 1, 'Business': 1, 'Scales': 1, 'of': 2, 'Tail': 1, "Lovers'": 1, 'Past': 1, 'Distant': 1, 'Happiness': 1, 'Partner': 1, 'Helpless': 1, 'Best': 1}
misc_words: ['Spice', 'Wolf', 'and']
-------------------------------------------------------------------------------------------------------------------------------------------------------------
URL: 'http://api.anidb.net:9001/httpapi?request=anime&client=hama&clientver=1&protover=1&aid=5406', Creating cached file: 'c:\users\admini~1\appdata\local\temp\2\ASS-anidb-5406.xml'
Error reading url 'http://api.anidb.net:9001/httpapi?request=anime&client=hama&clientver=1&protover=1&aid=5406', Exception: '<urlopen error [Errno 10013] An attempt was made to access a socket in a way forbidden by its access permissions>'
Error downloading 'http://api.anidb.net:9001/httpapi?request=anime&client=hama&clientver=1&protover=1&aid=5406', Exception: '<urlopen error [Errno 10013] An attempt was made to access a socket in a way forbidden by its access permissions>'

```

I attached the "Agent Update Log" for JJBA Golden Wind which is showing the missing episodes.
[[anidb-14173] JJBA Golden Wind.agent-update.log](https://github.com/ZeroQI/Hama.bundle/files/3269970/anidb-14173.JJBA.Golden.Wind.agent-update.log)

**Comment by ZeroQI** [comment]
titles: AniDB, TheTVDB | TheTVDB, AniDB means for series title (first part of pipe) AniDB then TheTVDB, but then for episodes (after the pipe) TheTVDB then AniDB, as AniDB doesn't have much episode names in other languages.
You may want to set titles: AniDB, TheTVDB | AniDB, TheTVDB

Not showing in Plex is a scanner issue
- Try to see it the folder for 'c:\users\admini~1\appdata\local\temp\2\ASS-anidb-5406.xml' have appropriate write permissions, that is probably the cause, no right access by plex
- If Plex is running under the system account, then it accesses the file system as such. The proper way is to pick a regular user account (or create one especially for Plex) and select it in windows services used by Plex

**Comment by sheik124** [comment]
Made the change for titles. Thank you.

**Comment by sheik124** [comment]
```
TVDBid: 346798 | Title: 'Asobi Asobase' | Missing Episode Summaries: ['s1e11', 's1e12']
TVDBid: 77435 | Title: 'Courage the Cowardly Dog' | Missing Episode Summaries: ['s2e16']
TVDBid: 358411 | Title: 'Hitoribocchi no Marumaruseikatsu' | Missing Episode Summaries: ['s1e11', 's1e12']
TVDBid: 360502 | Title: 'How Heavy Are the Dumbbells You Lift?' | Missing Episode Summaries: ['s1e3', 's1e4', 's1e5', 's1e6', 's1e7', 's1e8', 's1e9', 's1e10', 's1e11', 's1e12']
TVDBid: 357471 | Title: 'ISEKAI QUARTET' | Missing Episode Summaries: ['s1e12']
TVDBid: 262954 | Title: 'JoJo's Bizarre Adventure (2012)' | Missing Episode Summaries: ['s4e25', 's4e26', 's4e27', 's4e30', 's4e31', 's4e32', 's4e33', 's4e34', 's4e35', 's4e36', 's4e37', 's4e38', 's4e39']
TVDBid: 361707 | Title: 'Magical Sempai' | Missing Episode Summaries: ['s1e1', 's1e2', 's1e3', 's1e4', 's1e5', 's1e6', 's1e7', 's1e8', 's1e9', 's1e10', 's1e11', 's1e12']
TVDBid: 307375 | Title: 'Mob Psycho 100' | Missing Episode Summaries: ['s2e9', 's2e11', 's2e12', 's2e13']
TVDBid: 293088 | Title: 'One-Punch Man' | Missing Episode Summaries: ['s2e11', 's2e12']
TVDBid: 360579 | Title: 'The Helpful Fox Senko-san' | Missing Episode Summaries: ['s1e10', 's1e11', 's1e12']
```

Still haven't found a way to fix this.

Is there a way to brainwash AniDB and Hama _without_ deleting my entire library? Some kind of cache I can delete in `\Plex Media Server\Plug-in Support\Data\com.plexapp.agents.hama\`? When I checked the agent-update log for one of the shows (Golden Wind) it mentioned having cached local data in `TheTVDB/json/262954\series_en.json` so I moved that folder and tried again, still have the same blank episode titles and summaries.

**Comment by ZeroQI** [comment]
You gave me:
- hama logs for "JJBA Golden Wind"
- filelist and scanner logs for "Spice and Wolf [anidb-5406]"
- the season and ep number for "Seihou Bukyou outlaw star" s00e01 with issue wrong title

You are inconsistent and need to pick 1 exemple and follow it through:
- series hama update log
- season and episode
- describe issue

The update logs describe which field (title) it uses and which are available so you are clearly not understanding them as i can see issues logs provided "JJBA Golden Wind" which shows:
- tvdb part detected no episode matching
- mapping links anidb numbering to season 4 of tvdb series
- no metadata for any episode...
- anidb file download error for YOUR PC ONLY: '<urlopen error [Errno 10013] An attempt was made to access a socket in a way forbidden by its access permissions>'

**You need to open this url from the same machine** http://api.anidb.net:9001/httpapi?request=anime&client=hama&clientver=1&protover=1&aid=14173
- if you can't some spyware/firewall/antivirus prevent you opening on your end (port number 9001 forbidden?)
- if you can plex is refusedd access to the same url

AGAIN this is an issue on YOUR MACHINE!
If it cannot pull anidb info, it won't be available, I can't dumb it down more...
